tests/php/Model/SiteTreeTest_ExtensionA.php

```diff
@@ -13,4 +14,9 @@ public function canPublish($member)
     {
         return static::$can_publish;
     }
+
+    public function updateLink(&$link, $action = null)
+    {
+        $link = Controller::join_links($link, '?extra=1');
```

Comment:
I'd like to see a unit test prepending to the link, to cover what you're intending for this extension point as well.